runtime-modules/project-token/src/lib.rs

```diff
@@ -0,0 +1,648 @@
+
+        Ok(())
+    }
+
+    /// Returns the module account for the bloat bond treasury
+    pub fn bloat_bond_treasury_account_id() -> T::AccountId {
+        T::ModuleId::get().into_sub_account(Vec::<u8>::new())
+    }
+
+    pub(crate) fn validate_destination(
+        token_id: T::TokenId,
+        dst: T::AccountId,
+        transfer_policy: &TransferPolicyOf<T>,
+    ) -> Result<Validated<T::AccountId>, DispatchError> {
+        match (
+            transfer_policy,
+            Self::ensure_account_data_exists(token_id, &dst),
+        ) {
+            (&TransferPolicyOf::<T>::Permissionless, Err(_)) => {
+                Ok(Validated::<_>::NonExisting(dst))
+            }
+            (&TransferPolicyOf::<T>::Permissionless, Ok(_)) => Ok(Validated::<_>::Existing(dst)),
+            (&TransferPolicyOf::<T>::Permissioned(_), Ok(_)) => Ok(Validated::<_>::Existing(dst)),
+            (&TransferPolicyOf::<T>::Permissioned(_), Err(e)) => Err(e),
+        }
+    }
+
+    pub(crate) fn compute_bloat_bond(
+        validated_transfers: &ValidatedTransfers<T>,
+    ) -> ReserveBalanceOf<T> {
+        let bloat_bond = Self::bloat_bond();
+        validated_transfers
+            .iter()
+            .fold(ReserveBalanceOf::<T>::zero(), |acc, (account, _)| {
+                if matches!(account, Validated::<_>::NonExisting(_)) {
+                    acc.saturating_add(bloat_bond)
+                } else {
+                    ReserveBalanceOf::<T>::zero()
+                }
+            })
+    }
+
+    pub(crate) fn validate_transfers(
+        token_id: T::TokenId,
+        transfers: TransfersOf<T>,
+        transfer_policy: &TransferPolicyOf<T>,
+    ) -> Result<ValidatedTransfers<T>, DispatchError> {
+        let result = transfers
+            .into_iter()
+            .map(|(dst, payment)| {
+                Self::validate_destination(token_id, dst, transfer_policy).map(|res| (res, payment))
+            })
+            .collect::<Result<BTreeMap<_, _>, _>>()?;
+
+        Ok(Transfers::<_, _>(result))
+    }
+
+    pub(crate) fn ensure_can_transfer_reserve(
+        src: &T::AccountId,
+        dst: &T::AccountId,
+        amount: ReserveBalanceOf<T>,
+    ) -> DispatchResult {
+        if !amount.is_zero() {
+            let src_free = T::ReserveCurrency::free_balance(src);
+            let dst_free = T::ReserveCurrency::free_balance(dst);
+
+            ensure!(
+                src_free >= amount.saturating_add(T::ReserveExistentialDeposit::get()),
+                Error::<T>::InsufficientBalanceForBloatBond,
+            );
+
+            ensure!(
+                dst_free.saturating_add(amount) >= T::ReserveExistentialDeposit::get(),
+                Error::<T>::InsufficientBalanceForBloatBond,
+            );
+
+            T::ReserveCurrency::ensure_can_withdraw(
+                src,
+                amount,
+                WithdrawReason::Transfer.into(),
+                src_free,
```

Comment:
```suggestion
                src_free.saturating_sub(amount),
```
Notice that in `pallet-balances` there is `from_account.free = from_account.free.checked_sub(&value)` before:
```
				Self::ensure_can_withdraw(
					transactor,
					value,
					WithdrawReason::Transfer.into(),
					from_account.free,
				)?;
```

Reply:
fixed:  84d802bf60b8207789b28f5e5da04375a3f4c9fe